bson/primitive/decimal.go

```diff
@@ -211,6 +212,40 @@ func (d Decimal128) IsZero() bool {
 	return d.h == 0 && d.l == 0
 }
 
+// MarshalJSON returns Decimal128 as a string.
+func (d Decimal128) MarshalJSON() ([]byte, error) {
+	return json.Marshal(d.String())
+}
+
+// UnmarshalJSON creates a primitive.Decimal128 from a JSON string.
+func (d *Decimal128) UnmarshalJSON(b []byte) error {
```

Comment:
Seems like it's a common pattern to check for the string `null` so that JSON null values can be unmarshalled to non-pointer data types. The standard library [time.Time](https://golang.org/src/time/time.go?s=38690:38737#L1246) type does this and we mimic that behavior for [primitive.DateTime](https://github.com/mongodb/mongo-go-driver/blob/60f76f5b1810553090503950fce37c876253217b/bson/primitive/primitive.go#L52).

Reply:
done